lib/liquid/standardfilters.rb

```diff
@@ -391,6 +391,17 @@ def round(input, n = 0)
       raise Liquid::FloatDomainError, e.message
     end
 
+    # Defaults are passed as nil so systems can easily override
+    def format_number(input, options = {})
+      options = {} unless options.is_a?(Hash)
+      precision = options['precision'] || 2
+      delimiter = options['delimiter'] || " ".freeze
+      separator = options['separator'] || ".".freeze
+      return input if (prec = Utils.to_number(precision).to_i) < 0
+      whole_part, decimal_part = Kernel.format("%.#{prec}f", Utils.to_number(input)).split('.')
+      [whole_part.gsub(/(\d)(?=\d{3}+$)/, "\\1#{delimiter}"), decimal_part].compact.join(separator.to_s)
```

Comment:
The regex should be moved to a constant since it's static. Also we should use `\z` instead of `$` (even if `whole_part` can't contain new lines; as a Rubyist usage of `$` instantly raises a flag in my head when I read this code).